distribution/openrct2.d.ts

```diff
@@ -531,66 +534,88 @@ declare global {
     }
 
     interface FootpathElement extends BaseTileElement {
-        footpathType: number;
-        edgesAndCorners: number;
+        object: number;
+
+        edges: number;
+        corners: number;
         slopeDirection: number | null;
         isBlockedByVehicle: boolean;
         isWide: boolean;
 
         isQueue: boolean;
         queueBannerDirection: number | null;
-        ride: number;
-        station: number;
+        ride: number | null;
+        station: number | null;
 
         addition: number | null;
-        isAdditionBroken: boolean;
-
-        direction: Direction;
+        additionStatus: number | null;
+        isAdditionBroken: boolean | null;
+        isAdditionGhost: boolean | null;
     }
 
     interface TrackElement extends BaseTileElement {
+        direction: Direction;
         trackType: number;
-        sequence: number;
+        sequence: number | null;
+        mazeEntry: number | null;
+
+        colourScheme: number | null;
+        seatRotation: number | null;
+
         ride: number;
-        station: number;
+        station: number | null;
+
+        brakeBoosterSpeed: number | null;
         hasChainLift: boolean;
-        direction: Direction;
+        isInverted: boolean;
+        hasCableLift: boolean;
     }
 
     interface SmallSceneryElement extends BaseTileElement {
+        direction: Direction;
         object: number;
         primaryColour: number;
         secondaryColour: number;
-        direction: Direction;
         quadrant: number;
+        age: number;
     }
 
-    interface EntranceElement extends BaseTileElement {
+    interface WallElement extends BaseTileElement {
+        direction: Direction;
         object: number;
-        sequence: number;
-        ride: number;
-        station: number;
+        primaryColour: number;
+        secondaryColour: number;
+        tertiaryColour: number;
+        bannerIndex: number;
```

Comment:
bannerIndex could be null

Reply:
The way it is implemented right now, bannerIndex is never null. But bannerIndex == 0 is ambigious, as it can either mean no banner or the first banner.
I can change that by e.g. checking if GetBanner() == nullptr first.